File: src/test/java/io/binac/leetcode/TwoSumTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TwoSumTest {
    private final TwoSum.Solution1 solution1 = new TwoSum.Solution1();

    @Test
    void test1() {
        int nums[] = {2, 7, 11, 15};
        int result[] = solution1.twoSum(nums, 9);
        int expected[] = {0, 1};
        assertArrayEquals(expected, result);
    }
}

File: cpp/unittests/TwoSumTest.cpp
#include "TwoSum.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(TwoSum, Solution1_1) {
  std::vector<int> Nums{2, 7, 11, 15};
  const int Target = 9;
  const std::vector<int> Expected{0, 1};
  EXPECT_EQ(Expected, Solution1_1().twoSum(Nums, Target));
}